Add RHEL 10 s390x preferences and update tests

nekkunti opened the pull request:
- Added new `preferences/rhel/10/s390x` directory for RHEL 10 (s390x) instance types
- Updated `preferences/rhel/10/kustomization.yaml` to include s390x resources
- Updated `README.md` to list `rhel.10.s390x`
- Extended functional test coverage in `instancetype_test.go` to include `rhel.10.s390x` variant

  https://github.com/kubevirt/common-instancetypes/pull/361

<!--  Thanks for sending a pull request!  Here are some tips for you:
1. Follow the instructions for writing a release note from k8s: https://git.k8s.io/community/contributors/guide/release-notes.md
-->

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note

```

nekkunti commented:
> /hold /cc @jcanocan
> 
> Can you cherry-pick [462004b](https://github.com/kubevirt/common-instancetypes/commit/462004b541e4c7b41aa1170beaa947d42a92300a) to the release-1.2 branch, fix and document the conflicts in place instead of creating a standalone commit?

@lyarwood Done,  Please have a look

jcanocan commented:
Thanks @vamsikrishna-siddu for your PR!

Could you please edit the PR title as "[release-1.2] Add RHEL 10 s390x preferences and update tests"?
Moreover, could you please indicate in the PR description that this is a manual backport of https://github.com/kubevirt/common-instancetypes/pull/361? 

Here, you can find an example PR of a manual backport: https://github.com/kubevirt/common-instancetypes/pull/452. Hope you find it helpful.

Rather than that, the PR looks great!

jcanocan commented:
> > Thanks @vamsikrishna-siddu for your PR!
> > Could you please edit the PR title as "[release-1.2] Add RHEL 10 s390x preferences and update tests"? Moreover, could you please indicate in the PR description that this is a manual backport of #361?
> > Here, you can find an example PR of a manual backport: #452. Hope you find it helpful.
> > Rather than that, the PR looks great!
> 
> Thanks @jcanocan , I have updated

You are missing the release-note. Please, just gap the same exact format of #452, and adapt it to your case.

nekkunti commented:
> > > Thanks @vamsikrishna-siddu for your PR!
> > > Could you please edit the PR title as "[release-1.2] Add RHEL 10 s390x preferences and update tests"? Moreover, could you please indicate in the PR description that this is a manual backport of #361?
> > > Here, you can find an example PR of a manual backport: #452. Hope you find it helpful.
> > > Rather than that, the PR looks great!
> > 
> > 
> > Thanks @jcanocan , I have updated
> 
> You are missing the release-note. Please, just gap the same exact format of #452, and adapt it to your case.

Thanks  @jcanocan  , added release note , done same format as ref PR
